### ISSUE TITLE
Significant difference in trained model between tf2.3 and tf>2.4 

### ISSUE THREAD

**Issue by jviquerat** [tensorflow/tensorflow#52552]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): osx 11.2 and google colab
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.3/2.4/2.5/2.6
- Python version: 3.8.11

**Describe the current behavior**
I am working on a small deep q-network code for a course, applying it to cartpole-v0
The implementation achieves standard DQN performance level when using tf2.3.0, but learning fails (code runs but performance is extremely poor) when using tf2.4.0 and superior (tested 2.4.0, 2.5.0, 2.6.0 under osx, came accross the problem while trying to run the code on colab)

**Describe the expected behavior**
Performance should be identical on all versions

- Do you want to contribute a PR? (yes/no): no

**Standalone code to reproduce the issue**
This code runs fine with tf2.3 (it shows standard DQN performance level), while learning performance totally collapses for tf2.4 and superior:
https://colab.research.google.com/drive/1xKh_Lw8gd1chPhNC8MzXmifC2zguzgU_?usp=sharing

Thanks in advance

**Comment by jviquerat** [comment]
Just noticed that by removing the `clipnorm` option in `Adam`, things seem to get back to normal.
Could it be related to <a href="https://github.com/tensorflow/tensorflow/issues/36001">this issue</a> ? If yes, I'm not sure about the proper way of clipping gradients ?

**Comment by sachinprasadhs** [comment]
There was some changes made from Tensorflow 2.4 version which states as below.
`Optimizer.__init__ now accepts a gradient_aggregator to allow for customization of how gradients are aggregated across devices, as well as gradients_transformers to allow for custom gradient transformations (such as gradient clipping).`
May be this could be the reason, you can refer to [this](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md#tfkeras) 2.4 release note. Thanks.
